src/query/storages/parquet/src/parquet_rs/parquet_reader/read_policy/topk_only.rs

```diff
@@ -0,0 +1,216 @@
+            .filter(|f| f.name() != topk_field.name())
+            .collect::<Vec<_>>();
+        let remain_schema = TableSchema::new(remain_fields);
+        let remain_field_levels =
+            parquet_to_arrow_field_levels(schema_desc, remain_projection.clone(), None)?;
+        let remain_field_paths = Arc::new(compute_output_field_paths(
+            schema_desc,
+            &remain_projection,
+            &remain_schema,
+            inner_projection,
+        )?);
+
+        let mut src_schema = remain_schema;
+        src_schema.fields.push(topk_field);
+        let src_schema = Arc::new(DataSchema::from(&src_schema));
+        let dst_schema = Arc::new(DataSchema::from(output_schema));
+
+        Ok(Box::new(Self {
+            topk,
+            remain_projection,
+            remain_field_levels,
+            remain_field_paths,
+            src_schema,
+            dst_schema,
+        }))
+    }
+}
+
+#[async_trait::async_trait]
+impl ReadPolicyBuilder for TopkOnlyPolicyBuilder {
+    async fn build(
+        &self,
+        mut row_group: InMemoryRowGroup<'_>,
+        mut selection: Option<RowSelection>,
+        sorter: &mut Option<TopKSorter>,
+        batch_size: usize,
+    ) -> Result<Option<ReadPolicyImpl>> {
+        debug_assert!(sorter.is_some());
+        let sorter = sorter.as_mut().unwrap();
+        let mut num_rows = selection
+            .as_ref()
+            .map(|x| x.row_count())
+            .unwrap_or(row_group.row_count());
+
+        // Prefetch the topk column.
+        row_group
+            .fetch(self.topk.projection(), selection.as_ref())
+            .await?;
+        let mut reader = ParquetRecordBatchReader::try_new_with_row_groups(
+            self.topk.field_levels(),
+            &row_group,
+            num_rows, // Read all rows at one time.
+            selection.clone(),
+        )?;
+        let batch = reader.next().transpose()?.unwrap();
+        debug_assert!(reader.next().is_none());
+        // Topk column **must** not be in a nested column.
+        let prefetched = transform_record_batch(&batch, &None)?;
+        debug_assert_eq!(prefetched.num_columns(), 1);
+        let topk_col = prefetched.columns()[0].value.as_column().unwrap();
+        let filter = self.topk.evaluate_column(topk_col, sorter);
+        if filter.unset_bits() == num_rows {
+            // All rows are filtered out.
+            return Ok(None);
+        }
+        let prefetched = prefetched.filter_with_bitmap(&filter)?;
+        let filter = bitmap_to_boolean_array(filter);
+        let sel = RowSelection::from_filters(&[filter]);
+        // Update row selection.
+        match selection.as_mut() {
+            Some(selection) => {
+                *selection = selection.and_then(&sel);
+            }
+            None => {
+                selection = Some(sel);
+            }
+        }
+
+        // Slice the prefetched block by `batch_size`.
+        num_rows = prefetched.num_rows();
```

Comment:
This code looks the same as the implementation in `PredicateAndTopkPolicyBuilder`, can we write them in a single function to reduce duplication of code?
https://github.com/datafuselabs/databend/pull/13020/files#diff-11b6a41c2c804bda82298bb8115e50af179135d64aa13384f17ff6ffb05403ceR230-R252